pkgs/steam-jupiter/fhsenv.nix

```diff
@@ -0,0 +1,90 @@
+# A wrapped version of Steam with shims to satisfy the SteamOS-only
+# dependencies of the Steam Deck UI
+
+{ lib
+, runCommand
+, writeShellScriptBin
+, steam-fhsenv
+, ...
+} @ args:
+
+let
+  extraArgs = builtins.removeAttrs args [ "lib" "runCommand" "writeShellScriptBin" "steam-fhsenv" ];
+
+  # The sudo wrapper doesn't work in FHS environments. For our purposes
+  # we add a passthrough sudo command that does not actually escalate
+  # privileges.
+  #
+  # <https://github.com/NixOS/nixpkgs/issues/42117>
+  passthroughSudo = writeShellScriptBin "sudo" ''
+    declare -a final
+
+    positional=""
+    for value in "$@"; do
+      if [[ -n "$positional" ]]; then
+        final+=("$value")
+      elif [[ "$value" == "-n" ]]; then
+        :
+      else
+        positional="y"
+        final+=("$value")
+      fi
+    done
+
+    exec "''${final[@]}"
+  '';
+
+  # Null SteamOS updater that does nothing
+  #
+  # This gets us past the OS update step in the OOBE wizard.
+  nullOsUpdater = writeShellScriptBin "steamos-update" ''
+    >&2 echo "steamos-update: Not supported on NixOS - Doing nothing"
+    exit 7;
+  '';
+
+  # Null Steam Deck BIOS updater that does nothing
+  nullBiosUpdater = writeShellScriptBin "jupiter-biosupdate" ''
+    >&2 echo "jupiter-biosupdate: Doing nothing"
+  '';
+
+  # A very simplistic "session switcher." All it does is kill gamescope.
+  sessionSwitcher = writeShellScriptBin "steamos-session-select" ''
```

Comment:
Something to consider for later: how do we want to make this "hookable" for end-users?